plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p5edges/orthogonal/HyperEdgeSegment.java

```diff
@@ -271,6 +273,130 @@ public int getCriticalInWeight() {
+        return splitBy;
+    }
+    
+    /**
+     * Sets the segment that caused this one to be split, if any.
+     */
+    public void setSplitBy(final HyperEdgeSegment splitBy) {
+        this.splitBy = splitBy;
+    }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Utilities
+    
+    /**
+     * Convernience method which returns the size of this segment, which is its end coordinate minus its start
+     * coordinate.
+     */
+    public double getSize() {
+        return getEndCoordinate() - getStartCoordinate();
+    }
+    
+    /**
+     * Checks whether this segment connects two or more ports.
+     */
+    public boolean representsHyperedge() {
+        return getIncomingConnectionCoordinates().size() + getOutgoingConnectionCoordinates().size() > 2;
+    }
+    
+    /**
+     * Checks whether this segment was introduced while splitting another segment.
+     */
+    public boolean isDummy() {
+        return splitPartner != null && splitBy == null;
+    }
+    
+    /**
+     * Recomputes the start and end coordinate based on incoming and outgoing connection coordinates.
+     */
+    private void recomputeExtent() {
+        recomputeExtent(incomingConnectionCoordinates);
+        recomputeExtent(outgoingConnectionCoordinates);
+    }
+    
+    private void recomputeExtent(final LinkedList<Double> positions) {
+        // this code assumes that the positions are sorted ascendingly
+        if (!positions.isEmpty()) {
+            // set new start position
+            if (Double.isNaN(startPosition)) {
+                startPosition = positions.getFirst();
+            } else {
+                startPosition = Math.min(startPosition, positions.getFirst());
+            }
+            
+            // set new end position
+            if (Double.isNaN(endPosition)) {
+                endPosition = positions.getLast();
+            } else {
+                endPosition = Math.max(endPosition, positions.getLast());
+            }
+        }
+    }
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // Splitting
+    
+    /**
+     * Splits this segment into two and returns the new segment. The segments will be linked at the given position. This
+     * segment will retain all incoming connection coordinates, but all of its outgoing connection coordinates will move
+     * over to the new segment, to be replaced by the link between the two. This will completely clear all dependencies
+     * since they may have become obsolete.
+     * 
+     * @param splitPosition
+     *            position to split the two segments.
+     * @return the new segment for convenience, although the two will be linked as split partners.
+     */
+    public HyperEdgeSegment splitAt(final double splitPosition) {
+        splitPartner = new HyperEdgeSegment(routingStrategy);
```

Comment:
Below you used `.this`, here not. Any reasons?

Reply:
I was using `this` below to make it very clear where I'm setting properties on _this_ object and where I'm setting the according properties on the split partner.